File: backend/app/schemas/inventory.py
from datetime import datetime
from pydantic import BaseModel, Field


class InventoryCreate(BaseModel):
    product_id: int
    quantity: int = 0
    minimum_stock: int = 10
    maximum_stock: int = 100


class InventoryResponse(BaseModel):
    id: int
    product_id: int
    quantity: int
    minimum_stock: int
    maximum_stock: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AddStockRequest(BaseModel):
    quantity: int = Field(
        ...,
        gt=0,
        description="Quantity to add"
    )
    reference: str | None = Field(default=None, max_length=100)
    remarks: str | None = Field(default=None)


class LowStockResponse(BaseModel):
    product: str
    quantity: int

    class Config:
        from_attributes = True


class OutOfStockResponse(BaseModel):
    product: str

    class Config:
        from_attributes = True


class InventoryDashboardResponse(BaseModel):
    total_products: int
    total_stock: int
    low_stock_items: int
    out_of_stock_items: int

    class Config:
        from_attributes = True

File: frontend/types/inventory.ts
export interface InventoryCreate {
  product_id: number;
  quantity?: number;
  minimum_stock?: number;
  maximum_stock?: number;
}

export interface Inventory {
  id: number;
  product_id: number;
  quantity: number;
  minimum_stock: number;
  maximum_stock: number;
  created_at: string;
  updated_at: string;
}

export interface StockMutationRequest {
  quantity: number;
  reference?: string | null;
  remarks?: string | null;
}

export interface LowStockItem {
  product: string;
  quantity: number;
}

export interface OutOfStockItem {
  product: string;
}

export interface InventoryDashboard {
  total_products: number;
  total_stock: number;
  low_stock_items: number;
  out_of_stock_items: number;
}

export interface InventoryTransaction {
  id: number;
  product_id: number;
  transaction_type: string;
  quantity: number;
  reference?: string | null;
  remarks?: string | null;
  created_at: string;
  product_name?: string;
  sku?: string;
}

export interface InventoryItem {
  id: number;
  product_name: string;
  sku: string;
  current_stock: number;
  reorder_level: number;
  warehouse_name?: string | null;
  minimum_stock: number;
  maximum_stock: number;
}
